src/transformers/tokenization_tapas.py

```diff
@@ -0,0 +1,2666 @@
+                Predicted aggregation operator indices of the aggregation head. 
+        """
+        # compute probabilities from token logits
+        dist_per_token = torch.distributions.Bernoulli(logits=logits)
+        probabilities = dist_per_token.probs * data["attention_mask"].type(torch.float32).to(
+            dist_per_token.probs.device
+        )
+
+        token_types = [
+            "segment_ids",
+            "column_ids",
+            "row_ids",
+            "prev_label_ids",
+            "column_ranks",
+            "inv_column_ranks",
+            "numeric_relations",
+        ]
+
+        # collect input_ids, segment ids, row ids and column ids of batch. Shape (batch_size, seq_len)
+        input_ids = data["input_ids"]
+        segment_ids = data["token_type_ids"][:, :, token_types.index("segment_ids")]
+        row_ids = data["token_type_ids"][:, :, token_types.index("row_ids")]
+        column_ids = data["token_type_ids"][:, :, token_types.index("column_ids")]
+
+        # next, get answer coordinates for every example in the batch
+        num_batch = input_ids.shape[0]
+        predicted_answer_coordinates = []
+        for i in range(num_batch):
+            probabilities_example = probabilities[i].tolist()
+            segment_ids_example = segment_ids[i]
+            row_ids_example = row_ids[i]
+            column_ids_example = column_ids[i]
+
+            max_width = column_ids_example.max()
+            max_height = row_ids_example.max()
+
+            if max_width == 0 and max_height == 0:
+                continue
+
+            cell_coords_to_prob = self._get_mean_cell_probs(
+                probabilities_example,
+                segment_ids_example.tolist(),
+                row_ids_example.tolist(),
+                column_ids_example.tolist(),
+            )
+
+            # Select the answers above the classification threshold.
+            answer_coordinates = []
+            for col in range(max_width):
+                for row in range(max_height):
+                    cell_prob = cell_coords_to_prob.get((col, row), None)
+                    if cell_prob is not None:
+                        if cell_prob > cell_classification_threshold:
+                            answer_coordinates.append((row, col))
+            answer_coordinates = sorted(answer_coordinates)
+            predicted_answer_coordinates.append(answer_coordinates)
+
+        output = predicted_answer_coordinates
+
+        if logits_agg is not None:
+            predicted_aggregation_indices = logits_agg.argmax(dim=-1)
+            output = (output, predicted_aggregation_indices.tolist())
+
+        return output
+
+    #### End of everything related to converting logits to predictions ####
+
+
+""" BasicTokenizer and WordPieceTokenizer (taken from tokenization_bert.py)"""
+
+
+class BasicTokenizer(object):
```

Comment:
Can we add a comment `# Copied from transformers.models.bert.tokenization_bert.BasicTokenizer` if this is a perfect copy? That way our internal tooling will check the copy will not diverge from the original.